frontend/src/utils/giteaApi.js

```diff
@@ -170,21 +170,19 @@ export const getRepo = async fullname => {
 
 /**
  * Get all repos
- * @param csrf{string}
- * @returns {Promise<any|*[]>}
+ * @returns {Promise<[Object]>}
  */
-export const getRepos = csrf => searchRepos(csrf)
+export const getAllRepos = () => searchRepos()
 
 /**
  * Search all repos
- * @param csrf{string}
  * @param sort{string}
  * @param order{string}
- * @param q{string}: search query, leave undefined to return all repos
- * @returns {Promise<any|*[]>}
+ * @param q{string=}: search query, leave undefined to return all repos
+ * @returns {Promise<[Object]>}
  */
-export const searchRepos = async (csrf, sort = 'updated', order = 'desc', q) => {
-  const endpoint = `${giteaApiUrl}/repos/search?_csrf=${csrf}`
+export const searchRepos = async (sort = 'updated', order = 'desc', q) => {
+  const endpoint = `${giteaApiUrl}/repos/search`
```

Comment:
What's happening with the csrf? Why isn't it needed?

Reply:
See [CSRF with GET requests](https://security.stackexchange.com/questions/115794/should-i-use-csrf-protection-for-get-requests). Luckily, Gitea got this part right and it follows the right semantics of `GET` requests; it doesn't perform any action just get the publicly available data. All these endpoints don't even require authentication to get a response from them.